tests/component-tests/orator/js/OratorTests-Utils.js

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2015-2018 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+/* global fluid, jqUnit, sinon */
+
+(function () {
+    "use strict";
+
+    fluid.registerNamespace("fluid.tests.orator");
+
+    /*******************************************************************************
+     * DOM Reader MockTTS Grade
+     *******************************************************************************/
+
+    fluid.defaults("fluid.tests.orator.domReaderMockTTS", {
+        components: {
+            tts: {
+                type: "fluid.mock.textToSpeech",
+                options: {
+                    invokers: {
+                        // put back the orator's own queueSpeech method, but pass in the
+                        // mock queueSpeech function as the speechFn
+                        queueSpeech: {
+                            funcName: "fluid.orator.domReader.queueSpeech",
+                            args: ["{that}", "{that}.mockQueueSpeech", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                        },
+                        mockQueueSpeech: {
+                            funcName: "fluid.mock.textToSpeech.queueSpeech",
+                            args: ["{arguments}.0", "{that}.speechRecord", "{arguments}.1", "{arguments}.2", "{arguments}.3"]
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    /*******************************************************************************
+     * Sinon Stub Utils
+     *******************************************************************************/
+
+    fluid.defaults("fluid.tests.orator.stubs", {
+        members: {
+            stubs: {}
+        }
+    });
+
+    fluid.defaults("fluid.tests.orator.domReaderStubs", {
+        gradeNames: ["fluid.tests.orator.stubs"],
+        methods: ["play", "pause"],
+        listeners: {
+            "onCreate.addSpies": {
+                funcName: "fluid.tests.orator.addStubs",
+                priority: "first",
+                args: ["{that}.stubs", "{that}", "{that}.options.methods"]
+            },
+            "onDestroy.restore": {
```

Comment:
This namespace should be somehow complementary to the one added to onCreate. But if we are being destroyed, why do we restore something?

Reply:
The restore on destroy can be removed because we are just adding spies to our own methods.